helix-core/src/main/java/org/apache/helix/task/WorkflowContext.java

```diff
@@ -81,15 +85,20 @@ public void setJobState(String job, TaskState s) {
     if (states == null) {
       states = new TreeMap<String, String>();
       _record.setMapField(WorkflowContextProperties.JOB_STATES.name(), states);
+      markWorkflowContextAsModified();
```

Comment:
This is not necessary. If the block has been entered, it definitely will go through next block.

Reply:
Done.